src/rdd/rdd.rs

```diff
@@ -992,6 +994,68 @@ pub trait Rdd: RddBase + 'static {
         let min_fn = Fn!(|x: Self::Item, y: Self::Item| x.min(y));
         self.reduce(min_fn)
     }
+
+    fn top(&self, num: usize) -> Result<Vec<Self::Item>>
+    where
+        Self: Sized,
+        Self::Item: Data + Ord,
+    {
+        if num == 0 {
+            Ok(vec![])
+        } else {
+            let first_k_func = Fn!(move |partition: Box<dyn Iterator<Item = Self::Item>>|
+                -> Box<dyn Iterator<Item = BoundedMinPriorityQueue<Self::Item>>>  {
+                    let mut queue = BoundedMinPriorityQueue::new(num);
+                    partition.for_each(|item: Self::Item| queue.append(item));
+                    Box::new(std::iter::once(queue))
+            });
+
+            let queue = self
+                .map_partitions(first_k_func)
+                .reduce(Fn!(
+                    move |mut queue1: BoundedMinPriorityQueue<Self::Item>,
+                          queue2: BoundedMinPriorityQueue<Self::Item>|
+                          -> BoundedMinPriorityQueue<Self::Item> {
+                        queue1.merge(queue2);
+                        queue1
+                    }
+                ))?
+                .unwrap() as BoundedMinPriorityQueue<Self::Item>;
+
+            Ok(queue.into_vec_sorted())
```

Comment:
Since this is a final transformation, you can take self by move (this would play nicely with what I said about using `Into<Vec<T>>` this will allow you to avoid cloning altogether inside the method.